Sources/SwiftParser/Directives.swift

```diff
@@ -103,32 +109,62 @@ extension Parser {
       var loopProgress = LoopProgressCondition()
-      while let poundIfHandle = firstIteration ? self.canRecoverTo(.poundIfKeyword) : self.canRecoverTo(anyIn: IfConfigContinuationClauseStartKeyword.self)?.handle,
-        loopProgress.evaluate(self.currentToken)
-      {
-        var (unexpectedBeforePoundIf, poundIf) = self.eat(poundIfHandle)
-        firstIteration = false
-        // Parse the condition.
+      LOOP: while let (match, handle) = self.canRecoverTo(anyIn: PoundIfDirectiveKeywords.self), loopProgress.evaluate(self.currentToken) {
+        var unexpectedBeforePound: RawUnexpectedNodesSyntax?
+        var pound: RawTokenSyntax
         let condition: RawExprSyntax?
-        switch poundIf.tokenKind {
-        case .poundIfKeyword, .poundElseifKeyword:
+        var atElifTypo: Bool {
+          guard self.at(TokenSpec(.pound)), self.currentToken.trailingTriviaText.isEmpty else {
+            return false
+          }
+          let identifierSpec = TokenSpec(.identifier, allowAtStartOfLine: false)
+          var lookahead = self.lookahead()
+          lookahead.consumeAnyToken()
+          guard lookahead.at(identifierSpec), lookahead.currentToken.tokenText == "elif", lookahead.currentToken.leadingTriviaText.isEmpty else {
+            return false
+          }
+          lookahead.consumeAnyToken()
+          return lookahead.at(identifierSpec)
+        }
+
+        switch match {
+        case .poundIfKeyword:
+          if !firstIteration {
+            break LOOP
+          }
+          firstIteration = false
+          (unexpectedBeforePound, pound) = self.eat(handle)
+          condition = RawExprSyntax(self.parseSequenceExpression(.basic, forDirective: true))
+        case .poundElseifKeyword:
+          (unexpectedBeforePound, pound) = self.eat(handle)
           condition = RawExprSyntax(self.parseSequenceExpression(.basic, forDirective: true))
         case .poundElseKeyword:
+          (unexpectedBeforePound, pound) = self.eat(handle)
           if let ifToken = self.consume(if: .init(.if, allowAtStartOfLine: false)) {
-            unexpectedBeforePoundIf = RawUnexpectedNodesSyntax(combining: unexpectedBeforePoundIf, poundIf, ifToken, arena: self.arena)
-            poundIf = self.missingToken(.poundElseifKeyword)
+            unexpectedBeforePound = RawUnexpectedNodesSyntax(combining: unexpectedBeforePound, pound, ifToken, arena: self.arena)
+            pound = self.missingToken(.poundElseifKeyword)
             condition = RawExprSyntax(self.parseSequenceExpression(.basic, forDirective: true))
           } else {
             condition = nil
           }
-        default:
-          preconditionFailure("The loop condition should guarantee that we are at one of these tokens")
+        case .pound:
+          if atElifTypo {
+            (unexpectedBeforePound, pound) = self.eat(handle)
+            guard let identifier = self.consume(if: TokenSpec(.identifier, allowAtStartOfLine: false)) else {
```

Comment:
Nitpick to make it a little easier to read.

```suggestion
            guard let elif = self.consume(if: TokenSpec(.identifier, allowAtStartOfLine: false)) else {
```